pkg/api/componentreadiness/query/querygenerators.go

```diff
@@ -412,67 +439,90 @@ func getTestDetailsQuery(
 					FROM (%s) junit
 					INNER JOIN %s.jobs jobs ON junit.prowjob_build_id = jobs.prowjob_build_id
 					INNER JOIN latest_component_mapping cm ON testsuite = cm.suite AND test_name = cm.name
-`, client.Dataset, client.Dataset, fmt.Sprintf(dedupedJunitTable, jobNameQueryPortion, client.Dataset, junitTable, client.Dataset), client.Dataset)
+`, client.Dataset, client.Dataset, selectVariants, fmt.Sprintf(dedupedJunitTable, jobNameQueryPortion, client.Dataset, junitTable, client.Dataset), client.Dataset)
 
-	joinVariants := ""
-	for _, variant := range sortedKeys(allJobVariants.Variants) {
-		v := param.Cleanse(variant) // should be clean anyway, but just to make sure
-		joinVariants += fmt.Sprintf("LEFT JOIN %s.job_variants jv_%s ON variant_registry_job_name = jv_%s.job_name AND jv_%s.variant_name = '%s'\n",
-			client.Dataset, v, v, v, v)
-	}
 	queryString += joinVariants
 
-	groupString := `
+	groupString := fmt.Sprintf(`
 					GROUP BY
+						%s
 						file_path,
-						modified_time
+						modified_time,
+                        cm.id
 					ORDER BY
-						modified_time `
+						modified_time `, groupByVariants)
 	queryString += `
 					WHERE
 						(variant_registry_job_name LIKE 'periodic-%%' OR variant_registry_job_name LIKE 'release-%%' OR variant_registry_job_name LIKE 'aggregator-%%')
-						AND cm.id = @TestId `
-	commonParams := []bigquery.QueryParameter{
-		{
-			Name:  "TestId",
-			Value: c.TestIDOption.TestID,
-		},
+						AND
+`
+	commonParams := []bigquery.QueryParameter{}
+
+	for i, testIDOption := range testIDOpts {
+		queryString = addTestFilters(testIDOption, i, queryString, c, includeVariants, isSample)
+
+	}
+
+	if isSample {
+		queryString += filterByCrossCompareVariants(c.VariantOption.VariantCrossCompare, c.VariantOption.CompareVariants, &commonParams)
+	} else {
+		queryString += filterByCrossCompareVariants(c.VariantOption.VariantCrossCompare, includeVariants, &commonParams)
 	}
+	return queryString, groupString, commonParams
+}
+
+// addTestFilters injects query params to limit to one test and variants combo.
+func addTestFilters(
+	testIDOption crtype.RequestTestIdentificationOptions,
+	i int,
```

Comment:
nit: since this has been factored into a function of its own:
```suggestion
	index int,
```